File: backend-api/src/main/java/com/ogani/api/model/OrderDetail.java
package com.ogani.api.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "order_details")
public class OrderDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "detail_id")
    private Integer detailId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id")
    @JsonBackReference
    private Order order;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "category", "reviews", "order_details"})
    private Product product;

    private Integer quantity;

    @Column(name = "price_at_order", precision = 12, scale = 2)
    private BigDecimal priceAtOrder;

    @Column(precision = 12, scale = 2)
    private BigDecimal subtotal;
}

File: mobile_app/lib/models/order_detail.dart
import 'cart_item.dart';

class OrderDetail {
  final int detailId;
  final Product? product;
  final int quantity;
  final double priceAtOrder;
  final double subtotal;

  OrderDetail({
    required this.detailId,
    this.product,
    required this.quantity,
    required this.priceAtOrder,
    required this.subtotal,
  });

  factory OrderDetail.fromJson(Map<String, dynamic> json) {
    return OrderDetail(
      detailId: json['detailId'] ?? 0,
      product: json['product'] != null ? Product.fromJson(json['product']) : null,
      quantity: json['quantity'] ?? 0,
      priceAtOrder: (json['priceAtOrder'] ?? 0).toDouble(),
      subtotal: (json['subtotal'] ?? 0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'detailId': detailId,
      'product': product?.toJson(),
      'quantity': quantity,
      'priceAtOrder': priceAtOrder,
      'subtotal': subtotal,
    };
  }
}
